### PR TITLE
Use pytest to execute integration tests

### PR DESCRIPTION
**Pull request by pcerny**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import tests.config as test_config
+import tests.docker_utils as docker_utils
+import vertica_python
+
+
+configparser.ConfigParser()
+config_parser = configparser.RawConfigParser(allow_no_value=True)
+config_parser.read('pytest.ini')
+cfg: test_config.Config = test_config.Config.from_file(
+    config_parser.get('custom', 'config', fallback='tests/config_full.yaml')
```
Comment (jaceksan) [inline tests/conftest.py:16]: Default should be tests/config.yaml, not the full one.

File: README.md [hunk 2]
```diff
@@ -125,18 +125,19 @@ docker-compose up -d vertica
...
-It can be configured in [tests/config.yaml](tests/config.yaml).
+It can be configured in [tests/config.yaml](tests/config.yaml). Path to a config file can be changed in [pytest.ini](pytest.ini) file 
 For each required combination of OS / Vertica version the corresponding image is build, the container is started and tests are executed.
 All available customizations of build / run are applied and tested.
 
 Run tests:
 ```
 pip3 install requirements_tests.txt
-./tests.py
+pytest -vv tests
```
Comment (jaceksan) [inline README.md:135]: Document the need of installing python3-pytest.
Also on my ubuntu 20.04 default pytest is pytest-2, which obviously fails, so document here that pytest-3 is required.

Comment (pcerny) [reply README.md:135]: requirement file specifies that pytest must be in version 6.2.1. It is meant to be installed from PyPI, not by package in ubuntu or centos.

File: tests/conftest.py (hunk 1 above)
Comment (pcerny) [reply tests/conftest.py:16]: Fixed

Comment (pcerny) [reply tests/conftest.py:16]: Fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: README.md [hunk 3]
```diff
@@ -125,18 +125,19 @@ docker-compose up -d vertica
...
 # Optionally it is possible to test loading of VMART schema:
-./tests.py -l
+pytest -vv tests --test-vmart
 # Use different config file
-./tests.py -l -c tests/config_full.yaml
+# edit pytest.ini, section custom, key config
+pytest -vv tests
```
Comment (jaceksan) [inline README.md:140]: Document an option to copy pytest.ini e.g. into pytest_custom.ini and run:
```
pytest -c pytest_custom.ini -vv tests
```
Current way causes unstashed changes, which complicates life to a person, who develops changes into this repo.

File: tests/docker_utils.py [hunk 4]
```diff
@@ -2,55 +2,97 @@
 # -*- coding: utf-8 -*-
 # (C) 2021 GoodData Corporation
 
-from tests.cmd_utils import exec_cmd
-from time import time
+import re
+import time
+import requests
```
Comment (jaceksan) [inline tests/docker_utils.py:7]: Unused import

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/docker-image-for-vertica-fossa-scanning-zuul-el7/9/ : SUCCESS in 15s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/docker-image-for-vertica-merge-compliance-check/9/ : SUCCESS in 24s

**Pull request merged**